virtual_rainforest/core/grid.py

```diff
@@ -158,22 +162,19 @@ def make_hex_grid(
     prototype = scale(prototype, xfact=scale_factor, yfact=scale_factor, origin=(0, 0))
     prototype = translate(prototype, xoff=xoff, yoff=yoff)
 
-    # Tile the prototypes to create the grid
-    cell_list = [None] * (cell_nx * cell_ny)
-
-    # Loop over columns and rows of cells, incrementing the base coordinates
-    for y_idx in range(cell_ny):
-        for x_idx in range(cell_nx):
-            # Define the cell id as integer count starting lower left by row
-            cell_id = x_idx + (y_idx) * cell_nx
-            # Store shifted geometry under cell id
-            cell_list[cell_id] = translate(
-                prototype,
-                xoff=2 * apothem * x_idx + apothem * (y_idx % 2),
-                yoff=1.5 * side_length * y_idx,
-            )
+    # Get the cell ids and centres of the cells
+    idx_y, idx_x = np.indices((cell_ny, cell_nx))
+    cell_ids = idx_x + idx_y * cell_nx
```

Comment:
I can see how this could be used but - although these get flattened in the end - to start with I want `idx_y` and `idx_x` to retain the shape of `(cell_ny, cell_nx)`, because I then use `np.flipud` to get the direction of the y axis correctly. Am I missing something?

That 'help' page is a spectacular example of mystifying  `numpy` documentation 😄

Reply:
Hmm... I'm not sure `numpy.unravel_index` is what I was looking for after all. I think it's fine as is.